cdap-ui/app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsDetailsActions/PipelineDetailsActionsButton/index.js

```diff
@@ -0,0 +1,194 @@
+    existingSuffix = version;
+    version = version.replace('_v', '');
+    version = '_v' + ((!isNaN(parseInt(version, 10)) ? parseInt(version, 10) : 1) + 1);
+  } else {
+    version = '_v1';
+  }
+  return name.split(existingSuffix)[0] + version;
+};
+
+export default class PipelineDetailsActionsButton extends Component {
+  static propTypes = {
+    pipelineName: PropTypes.string,
+    description: PropTypes.string,
+    artifact: PropTypes.object,
+    config: PropTypes.object
+  };
+
+  state = {
+    showExportModal: false,
+    showDeleteConfirmationModal: false
+  };
+
+  pipelineConfig = {
+    name: this.props.pipelineName,
+    description: this.props.description,
+    artifact: this.props.artifact,
+    config: this.props.config
+  };
+
+  duplicateConfigAndNavigate = () => {
+    let bumpedVersionName = getClonePipelineName(this.props.pipelineName);
+    let pipelineConfigWithBumpedVersion = {...this.pipelineConfig, name: bumpedVersionName};
+    window.localStorage.setItem(bumpedVersionName, JSON.stringify(pipelineConfigWithBumpedVersion));
+    const hydratorLink = window.getHydratorUrl({
+      stateName: 'hydrator.create',
+      stateParams: {
+        namespace: getCurrentNamespace(),
+        cloneId: bumpedVersionName,
+        artifactType: this.props.artifact.name
+      }
+    });
+    window.location.href = hydratorLink;
+  };
+
+  deletePipeline = () => {
+    let namespace = getCurrentNamespace();
+    let params = {
+      namespace,
+      appId: this.props.pipelineName
+    };
+    const pipelinesListLink = window.getHydratorUrl({
+      stateName: 'hydrator.list',
+      stateParams: {
+        namespace
+      }
+    });
+
+    MyAppApi.delete(params)
+      .subscribe(() => {
+        this.setState({
+          deleteErrMsg: '',
+          extendedDeleteErrMsg: ''
+        });
+        window.location.href = pipelinesListLink;
+      }, (err) => {
+        this.setState({
+          deleteErrMsg: 'There was a problem with the pipeline you were trying to delete',
+          extendedDeleteErrMsg: err
+        });
+      });
+  }
+
+  toggleExportModal = () => {
+    this.setState({ showExportModal: !this.state.showExportModal });
+  }
+
+  toggleDeleteConfirmationModal = () => {
+    this.setState({
+      showDeleteConfirmationModal: !this.state.showDeleteConfirmationModal,
+      deleteErrMsg: '',
+      extendedDeleteErrMsg: ''
+    });
+  }
+
+  renderExportPipelineModal() {
+    if (!this.state.showExportModal) {
+      return null;
+    }
+
+    return (
+      <PipelineExportModal
+        isOpen={this.state.showExportModal}
+        onClose={this.toggleExportModal}
+        pipelineConfig={this.pipelineConfig}
+      />
+    );
+  }
+
+  renderDeleteConfirmationModal() {
+    if (!this.state.showDeleteConfirmationModal) {
+      return null;
+    }
+
+    const confirmationElem = () => <div>Are you sure you want to delete <strong><em>{this.props.pipelineName}</em></strong>?</div>;
```

Comment:
I believe there was a change to this confirmation message when this pipeline is trigger other pipeline